docs/interfaces.md

```diff
@@ -0,0 +1,571 @@
+        oneof request {
+                GetMetadataRequest metadata     = 1; 
+                validateMSPRequest validate_msp = 2;
+        }
+}
+
+message SessionRequeatResponse {
+        SessionRequest request = 1;
+        oneof response {
+                GetMetadataResponse metadata     = 1; 
+                validateMSPResponse validate_msp = 2; 
+        }
+}
+
+```
+
+
+# TLCC_Enclave
+
+## Interface:
+Defined in the TLCC EDL.
+```c++
+// TBD is this interface already sufficient to support multi-channel?
+// TODO check if a block contatains any channel information?
+
+// creates a new channel
+public int ecall_join_channel(
+        [in, size=gen_len] uint8_t *genesis, uint32_t gen_len);
+
+// pushes next block to tlcc
+public int ecall_next_block(
+        [user_check] uint8_t *block_bytes, uint32_t block_size);
+```
+
+```c++
+// this secure session functions are defined in a secure channel module
+//included from secure_channel.edl
+
+public int session_setup(
+        [in] setup SessionSetup,
+        [out] SessionSetupResponse);
+
+public int session_close(
+        [in] close SessionClose
+        [out] SessionCloseReponse);
+
+public int session_request(
+        [in] request SessionRequest,
+        [out] SessionRequestResponse;
+```
+
+```c++
+// addtionally we may use the state sealing module to persists TLCC state
+// see below the data sealing module.
+include sealed_storage.edl
+```
+
+Session requests are depatched to the following interfaces:
+```c++
+// TODO refine
+// verify state data
+public int get_state_metadata(
+        [in, string] const char *namespace,
+        [in, string] const char *key,
+        [in, size=32] uint8_t *nonce,
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+public int get_multi_state_metadata(
+        [in, string] const char *namespace,
+        [in, string] const char *comp_key,
+        [in, size=32] uint8_t *nonce,
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+// verify msp identities
+public int validate_msp_id(
+        [in, string] const char *msp_id,
+        [in, size=32] uint8_t *nonce,
+        [out] bool isValid, 
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+// verify msp identities
+public int can_endorse(
+        [in, string] const char *msp_id,
+        [in, size=32] uint8_t *nonce,
+        [out] bool isValid, 
+        [out] sgx_cmac_128bit_tag_t *cmac);
+```
+
+## State:
+The internal TLCC state consists of the current ledger state (block number, integrity-metadata, and msp) and sessions information for ecc communication.
+
+```c++
+// TODO: this needs some more brain cycles with respect of #402 and #410
+typedef struct internal_tlcc_state_t {
+        uint32_t sequence_number, // block number counter
+
+        // integrity metadata
+        kvs_t metadata_state,
+        
+        // keep ercc and lifecycle separetely as TLCC provides
+        // additionality, such as `can_endorse`, requiring the information
+        kvs_t ercc,
+        kvs_t lifecycle
```

Comment:
does lifecycle also cover the default policies? (TLCC would need the lifecycle policy to validate ERCC cc-def changes ...)